src/shell/output.c

```diff
@@ -292,15 +307,21 @@ static int shell_output_header_append (flux_shell_t *shell, json_t *output)
                                "stdout", "base64",
                                "stderr", "base64",
                              "count",
-                               "stdout", shell->info->jobspec->task_count,
-                               "stderr", shell->info->jobspec->task_count,
+                               "stdout", out->shell->info->jobspec->task_count,
+                               "stderr", out->shell->info->jobspec->task_count,
                              "options");
-    if (!o || json_array_append_new (output, o) < 0) {
-        json_decref (o);
+    if (!o) {
         errno = ENOMEM;
-        return -1;
+        goto error;
     }
-    return 0;
+    if (shell_output_commit (out, o) < 0)
```

Comment:
Now that we have the guest eventlog event, header can go out in the same commit as the first data entry.